lychee-bin/tests/cli.rs

```diff
@@ -786,11 +786,11 @@ mod cli {
         // run again to verify cache behavior
         test_cmd
             .assert()
-            .stdout(contains(format!(
+            .stderr(contains(format!(
                 "[200] {}/ | Cached: OK (cached)\n",
                 mock_server_ok.uri()
```

Comment:
I would like to get a second opinion on this part:
does it make sense to output `[200]` to `stderr`?
@thomas-zahner?

Reply:
Yes I think it makes sense, as these are just the status/progress report messages.
It's just the final result (JSON, MarkDown, etc.) that should be printeted to stdout, when lychee finished checking the links. (even if the result contains broken links)

This way the output can be used in much more useful ways, e.g. piping the JSON result into `jq`, writing or appending the MarkDown result to files with `>` or `>>` while the prigress is still separate.

I was also a bit unsure at some point whether this is really best-practice but this stackexchange discussion makes it quite clear: https://unix.stackexchange.com/questions/331611/do-progress-reports-logging-information-belong-on-stderr-or-stdout

Reply:
Perfect. Thanks a lot for your feedback and the research.